lib/extensions.js

```diff
@@ -776,4 +778,34 @@ module.exports = function(realmConstructor, context) {
         },
         configurable: true
     });
+
+    Object.defineProperty(realmConstructor, "JsonSerializationReplacer", {
+        get: function () {
+            const seen = [];
+
+            return function(_, value) {
+                if (typeof value !== "object") {
+                    return value;
+                }
+
+                const parent = this;
+                if (seen.length > 0) {
+                    const pos = seen.indexOf(parent);
+                    if (pos !== -1) {
+                        seen.splice(pos + 1)
```

Comment:
```suggestion
                        seen.splice(pos + 1);
```

Reply:
I think we should add `"semi": [1, "always"]` to the eslint config then.